paddle/gserver/gradientmachines/MultiGradientMachine.cpp

```diff
@@ -813,7 +813,6 @@ void TrainerThread::mergeGradSparse(
       para->getMat(PARAMETER_GRADIENT).get());
   std::vector<uint32_t>& ids = mainMat->getIds(threadId_);
 
-  ids.clear();
```

Comment:
This moved to ThreadParameterUpdater.cpp:233. Clearing gradient should be the job of ParameterUpdater